pkg/languagedetection/detector.go

```diff
@@ -36,34 +36,43 @@ type languageFromCLI struct {
 	validator func(exe string) bool
 }
 
-// rubyPattern is a regexp validator for the ruby prefix
-var rubyPattern = regexp.MustCompile(`^ruby\d+\.\d+$`)
+var (
+	rubyPattern = regexp.MustCompile(`^ruby\d+\.\d+$`)
+	phpPattern  = regexp.MustCompile(`^php\d$`)
```

Comment:
```suggestion
	phpPattern  = regexp.MustCompile(`^php(?:-fpm)?\d(?:\.\d)?$`)
```
`php8.1` and such is also a common name given to php executables.
Similarly, we need to cover `php-fpm8.1` as well.

Reply:
SGTM

Reply:
I just see the other comment, but I think it's reasonable to match all the relevant cases here.